enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/RunInClusterAction.java

```diff
@@ -0,0 +1,281 @@
+                            .build());
+
+                    client.apps()
+                            .deployments()
+                            .inNamespace(DEFAULT)
+                            .withName(projectName)
+                            .edit(d -> new DeploymentBuilder(d)
+                            .editSpec()
+                            .editTemplate()
+                            .editMetadata()
+                            .addToAnnotations("kubectl.kubernetes.io/restartedAt", Instant.now().toString()) //NOI18N
+                            .endMetadata()
+                            .endTemplate()
+                            .endSpec()
+                            .build());
+
+                } else {
+                    Deployment newDeployment = new DeploymentBuilder()
+                            .withNewMetadata()
+                            .withName(projectName)
+                            .addToLabels(APP, projectName)
+                            .endMetadata()
+                            .withNewSpec()
+                            .withReplicas(3)
+                            .withNewSelector()
+                            .addToMatchLabels(APP, projectName)
+                            .endSelector()
+                            .withNewTemplate()
+                            .withNewMetadata()
+                            .addToLabels(APP, projectName)
+                            .endMetadata()
+                            .withNewSpec()
+                            .addNewImagePullSecret().withName("docker-bearer-vscode-generated-ocirsecret").endImagePullSecret() //NOI18N
+                            .addNewContainer()
+                            .withName(projectName)
+                            .withImage(context.getUrl())
+                            .addNewPort()
+                            .withContainerPort(8080)
+                            .endPort()
+                            .endContainer()
+                            .endSpec()
+                            .endTemplate()
+                            .endSpec()
+                            .build();
+
+                    client.apps()
+                            .deployments()
+                            .inNamespace("default")
+                            .resource(newDeployment)
+                            .create();
+                }
+            });
+        } catch (InterruptedException | ExecutionException ex) {
+            Exceptions.printStackTrace(ex);
+        } finally {
+            h.finish();
+        }
+    }
+
+    private void runWithClient(ClusterItem cluster, Consumer<KubernetesClient> consumer) {
+        Config config = prepareConfig(cluster.getConfig(), cluster);
+        try (KubernetesClient client = new KubernetesClientBuilder().withConfig(config).build();) {
+            consumer.accept(client);
+        } catch (Exception e) {
+            Exceptions.printStackTrace(e);
```

Comment:
Swallows an exception -- the action should somehow inform the user.

Reply:
Will be addressed in subsequent PRs